src/utils/detectionApi.ts

```diff
@@ -359,32 +358,33 @@ class DetectionMapper {
         externalId: change.update.externalId,
       },
     };
-    (eventPatch as EventPatchHack).update.subType = change.update.label;
+    (eventPatch as EventPatchHack).update.subtype = change.update.label;
     if (change.update.metadata) {
-      Object.assign(eventPatch.update.metadata, change.update.metadata);
```

Comment:
The reason for using Object.assign here was to get a proper copy. I think the mapper might be working on the object that is coming in, and we don't want that to be reflected in the input object.